Composer/ScriptHandler.php

```diff
@@ -282,8 +282,10 @@ protected static function executeCommand(Event $event, $consoleDir, $cmd, $timeo
             $console .= ' --ansi';
         }
 
-        $process = new Process($php.($phpArgs ? ' '.$phpArgs : '').' '.$console.' '.$cmd, null, null, null, $timeout);
-        $process->run(function ($type, $buffer) use ($event) { $event->getIO()->write($buffer, false); });
+        $process = new Process($php.($phpArgs ? ' '.$phpArgs : '').' '.$console.' '.$cmd, null, $env, null, $timeout);
```

Comment:
Faced same issue today, but harder:
`ScriptHandler` completely ignores `ENV` and if there is a custom login (based on ENV) in kernel, eg. `AppKernel`, then '$_ENV' in child process would be empty and app flow is affected.

Suggest to pass `ENV` from parent to child procs:

```php
$env = array_merge($_ENV, (array) $env);
```